### PR TITLE
WIP: Component-specific aliases for variables

### PR DESCRIPTION
**Pull request by igloosi**
Allow for easier theming / general overriding of component variables.

Addresses: https://github.com/alphagov/govuk-frontend/issues/641, some of https://github.com/alphagov/govuk-frontend/issues/640

### CODE REVIEW

File: src/components/back-link/_back-link.scss [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 @include govuk-exports("back-link") {
 
+  $backlink-border-colour: $govuk-black !default;
```
Comment (nickcolley) [inline src/components/back-link/_back-link.scss]: Should these be namespaced with `govuk`?

Comment (36degrees) [reply src/components/back-link/_back-link.scss]: If we're expecting them to be referenced from outside of the component (e.g. by users customising the colours of a button before importing Frontend) then I think they probably should.

Is it worth holding off on doing this work until we've established our public API?

Comment (nickcolley) [reply src/components/back-link/_back-link.scss]: If these are not part of the public api we should consider not allowing them to be overridden?

Comment (igloosi) [reply src/components/back-link/_back-link.scss]: i'll stop then

**Pull request closed without merging**